### PR TITLE
Allow filtering MRT routes by IP version and overriding next-hop

### PR DESCRIPTION
**Pull request by XANi**
As title says, it adds few handy options to import.

I'm not really sure what use `mrtOpts` was supposed to have, as only a `Best` bool was used out of it, so I repurposed it to type usedpass arguments to injectMrt

### CODE REVIEW

File: gobgp/cmd/common.go [hunk 1]
```diff
@@ -108,10 +108,12 @@ var actionOpts struct {
 	NexthopAction       string `long:"next-hop" description:"specifying a next-hop action of policy"`
 }
 
-var mrtOpts struct {
+type mrtOpts struct {
```
Comment (fujita) [inline gobgp/cmd/common.go]: I prefer to keep 'var' for consistency. You can use 'mrtOpts' variant globally. The way to use neighborsOpts in gobgp/cmd/neighbor.go would be helpful.

File: gobgp/cmd/mrt.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func injectMrt(filename string, count int, skip int, onlyBest bool) error {
+func injectMrt(filename string, count int, skip int, opts *mrtOpts) error {
```
Comment (fujita) [inline gobgp/cmd/mrt.go]: You don't need to pass opts *mrtOpts if you keep 'var' above.

File: gobgp/cmd/mrt.go [hunk 3]
```diff
@@ -107,7 +109,7 @@ func injectMrt(filename string, count int, skip int, onlyBest bool) error {
 					paths = append(paths, table.NewPath(source, nlri, false, e.PathAttributes, t, false))
 				}
 
-				if onlyBest {
+				if opts.Best {
```
Comment (fujita) [inline gobgp/cmd/mrt.go]: mrtOpts.Best can be used here.

File: gobgp/cmd/mrt.go [hunk 4]
```diff
@@ -152,6 +154,7 @@ func injectMrt(filename string, count int, skip int, onlyBest bool) error {
 }
 
 func NewMrtCmd() *cobra.Command {
+	var opts mrtOpts
```
Comment (fujita) [inline gobgp/cmd/mrt.go]: You don't need it.

File: gobgp/cmd/mrt.go [hunk 5]
```diff
@@ -174,7 +177,7 @@ func NewMrtCmd() *cobra.Command {
 					}
 				}
 			}
-			err := injectMrt(filename, count, skip, mrtOpts.Best)
+			err := injectMrt(filename, count, skip, &opts)
```
Comment (fujita) [inline gobgp/cmd/mrt.go]: I guess that the original code confused you. You don't need to pass anything here. injectMrt() can access to mrtOpts variant.

Comment (XANi) [reply gobgp/cmd/mrt.go]: I thought it would be more readable if all parameters (well, all related to cli input) that change function is acting on would be explictly passed to it.

I changed it in the first place because originally my code was just
  ```go  
    func injectMrt(filename string, count int, skip int, skipV4 bool, skipV6, onlyBest bool)
```
but then I thought if there will be some other filter added in the future then that would be pretty ugly way to pass extra option so I moved it into mrtOpts

File: gobgp/cmd/common.go (hunk 1 above)
Comment (XANi) [reply gobgp/cmd/common.go]: Sure. Want me to add rest of the options there (filename, count, skip) ? Having half of options passed and half of it taken from some package variable would seem kinda hard to follow IMO.

Comment (fujita) [reply gobgp/cmd/common.go]: Surely, let's add them there and handle in the same way. Thnaks!

**Comment by XANi** [comment]
Done, I've also added option to change queue size and dropped it to some manageable value.

at current value of  it can eat 15 GB of RAM if importing [full view](http://data.ris.ripe.net/rrc21/) without `--only-best`

**Comment by fujita** [comment]
Sorry about the delay, pushed, thanks a lot!
Having changing 'var' to 'type' and then revering in the history doesn't make sense so I squashed the first three commits.
